src/peft/tuners/reft/config.py

```diff
@@ -0,0 +1,106 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+
+from peft.tuners.lycoris_utils import LycorisConfig
+from peft.utils import PeftType
+
+@dataclass
+class ReftConfig(LycorisConfig):
+    """
+    This is the configuration class to store the configuration of a [`REFTModel`].
+
+    Args:
+        r (`int`): REFT rank.
+        module_dropout (`int`): The dropout probability for disabling REFT modules during training.
+        target_modules (`Optional[Union[List[str], str]]`):
+            The names of the modules to apply the adapter to. If this is specified, only the modules with the specified
+            names will be replaced. When passing a string, a regex match will be performed. When passing a list of
+            strings, either an exact match will be performed or it is checked if the name of the module ends with any
+            of the passed strings. If this is specified as 'all-linear', then all linear modules are chosen, excluding
+            the output layer. If this is not specified, modules will be chosen according to the model architecture. If
+            the architecture is not known, an error will be raised -- in this case, you should specify the target
+            modules manually.
+        init_weights (`bool`):
+            Whether to perform initialization of REFT weights.
+        layers_to_transform (`Union[List[int], int]`):
+            The layer indices to transform. If a list of ints is passed, it will apply the adapter to the layer indices
+            that are specified in this list. If a single integer is passed, it will apply the transformations on the
+            layer at this index.
+        layers_pattern (`str`):
+            The layer pattern name, used only if `layers_to_transform` is different from `None`.
+        rank_pattern (`dict`):
+            The mapping from layer names or regexp expression to ranks which are different from the default rank
+            specified by `r`.
+        modules_to_save (`List[str]`):
+            List of modules apart from adapter layers to be set as trainable and saved in the final checkpoint.
+        block_share (`bool`):
+            Whether to share the REFT parameters between blocks or not. This is `False` by default.
+    """
+    r: int = field(default=8, metadata={"help": "REFT rank"})
+    alpha: int = field(default=8, metadata={"help": "Reft alpha"})
+    loc: Optional[Union[List[int], int]] = field(default=None, metadata={"help": "token locations are applied ReFT"})
```

Comment:
These two don't have a docstring entry. Also, the description "token locations are applied ReFT" needs to be explained better.